pkg/authentication/user_account.go

```diff
@@ -15,3 +19,22 @@ type UserAccount struct {
 	ZoneInfo    string    `json:"zoneinfo" db:"zone_info"`
 	HasLoggedIn bool      `json:"hasLoggedIn" db:"has_logged_in"`
 }
+
+type ctxKey string
+
+const (
+	userAccountServiceKey = ctxKey("userAccountService")
+)
+
+// GetUserAccountFromDBFunc represents a type of function which takes a context and a user_id uuid and returns a UserAccount
+type GetUserAccountFromDBFunc func(ctx context.Context, id uuid.UUID) (*UserAccount, error)
+
+// UserAccountService returns a GetUserAccountFromDBFunc that is decorating the context
+func UserAccountService(ctx context.Context) GetUserAccountFromDBFunc {
+	return ctx.Value(userAccountServiceKey).(GetUserAccountFromDBFunc)
+}
+
+// CTXWithUserAccountService decorates the context with a GetUserAccountFromDBFunc
+func CTXWithUserAccountService(ctx context.Context, accountFunction GetUserAccountFromDBFunc) context.Context {
```

Comment:
You know I think you are right, it honestly might be more consistent to leave out CTX or Context at all. Most places it happens in context.go in appcontext. I moved it into the other package when I was trying to avoid import cycle issues. 

I think I'll 
a. Check can this be moved to appcontext? If so, move it
b. depending on the package it needs to be in, name it consistently (likely remove CTX or Context)

Reply:
Thanks @OddTomBrooks Tom, there wasn't any issue moving this into the appcontext package, and I think that is a more correct place to include it. I also renamed it to be consistent with how it is named in that package!

Reply:
Awesome! Great work